### ISSUE TITLE
Arbitrary properties in .npmrc causes issues with other consumers

### ISSUE THREAD

**Issue by antony** [pnpm/pnpm#6656]
Adding arbitrary properties into .npmrc is problematic

Specifically, this breaks deploying to Vercel if we want pnpm to control the node version. Vercel validates this file:

`Detected unsupported "use-node-version" in your ".npmrc". Please use "engines" in your "package.json" instead.`

I noticed this issue https://github.com/pnpm/pnpm/issues/6309 which seems like it would be a good resolution to this issue.

I think that this specific issue aside, adding properties to .npmrc has a risk of future `npm` defined properties conflicting.

### pnpm version:

v8.5.0

### Code to reproduce the issue:

```bash
// .npmrc
use-node-version=xxx
```

### Additional information:

 - `node -v` prints:

v19.8.1

 - Windows, macOS, or Linux?:

Linux aive 6.0.12-76060006-generic #202212290932~1673966427~22.04~e15d5b5 SMP PREEMPT_DYNAMIC Tue J x86_64 x86_64 x86_64 GNU/Linux

**Comment by zkochan** [comment]
I agree that we need a separate config. It would be also great to write our own config reader instead of the old npm fork that is used now. But the easiest and safest option for now would be probably to read from a `.pnpmrc` ini file.

**Comment by antony** [comment]
I wonder if it would make sense to continue using a yaml file, rather than a .pnpmrc. Is there a specfic desire for the .ini format?

I also think a quick fix for this is to just prefer reading .pnpmrc first, is there a specific aversion to the forked code you're currently using?

as an aside, my immediate issue would be better solved by this https://github.com/pnpm/pnpm/issues/7211

**Comment by antony** [comment]
The latest version of `npm` is now writing explicit warnings about pnpm directives in `.npmrc` (which don't really matter, but could be disconcerting to some users)

```
npm warn Unknown project config "use-node-version". This will stop working in the next major version of npm.
npm warn Unknown project config "node-version". This will stop working in the next major version of npm.
npm warn Unknown project config "release" (node-mirror:release). This will stop working in the next major version of npm.
```

**Comment by zkochan** [comment]
This no longer reproduces on current pnpm, so closing.

pnpm reads its own settings from pnpm-workspace.yaml now, so nothing pnpm-specific has to go into .npmrc.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
